python/holidays/is-IS.py

```diff
@@ -0,0 +1,73 @@
+from dateutil.easter import EASTER_WESTERN
+
+from utils import SmartDayArrow
+from .holidays import Holiday, Locale
+
+
+class is_IS(Locale):
+    """
+    01-01: [NRF] Nýársdagur
+    3 days before Easter: [NRV] Skírdagur
+    2 days before Easter: [NRV] Föstudagurinn langi
+    Easter: [NRV] Páskadagur
+    1 day after Easter: [NRV] Annar dagur páska
+    39 days after Easter: [NRV] Uppstigningardagur
+    49 days after Easter: [NRV] Hvítasunnudagur
+    50 days after Easter: [NRV] Annar dagur hvítasunnu
+    05-01: [NF] Verkalýðsdagurinn
+    06-17: [NF] þjóðhátíðardagurinn
+    1. monday in August: [NV] Frídagur verslunarmanna
+    12-25: [NRF] Jóladagur
+    12-26: [NRF] Annar dagur jóla
+    """
+
+    locale = "is-IS"
+    easter_type = EASTER_WESTERN
+
+    def holiday_first_day_of_summer(self):
+        """
+        Calculate sumardagurinn fyrsti (first day of summer).
+
+        The holiday falls on the first Thursday after the 18th of April.
+        """
+        return [
+            Holiday(
+                locale=self.locale,
+                region="",
+                date=SmartDayArrow(self.year, 4, 18).shift_to_weekday("thursday"),
+                description="Sumardagurinn fyrsti",
+                flags="NV",
+                notes="",
+            )
+        ]
+
+    def holiday_half_days(self):
+        """
+        Define half-day holidays.
+
+        Both Christmas Eve (_aðfangadagur_) and New Year's Eve
+        (_gamlársdagur_) are public holidays in Iceland from 13:00 only.
+        They're included as full-day holidays, but with an explanatory
+        note.
+
+        12-24: [NRF] Aðfangadagur
+        12-31: [NRF] Gamlársdagur
```

Comment:
"Gamlársdagur" is not mentioned among the holidays in "Lög um frið vegna helgihalds" therefore I would flag it as "[NF]" only

Reply:
Fixed in https://github.com/flother/holidata/commit/9f888ed6ce5228fe73ff10b583fe3192f03ca09f.